events/api/src/test/java/org/projectnessie/events/api/TestJsonSerde.java

```diff
@@ -0,0 +1,373 @@
+            .putProperty("number", 123)
+            .putProperty("boolean", true)
+            .build();
+    assertThat(deserialize(serialize(event), Event.class)).isEqualTo(event);
+  }
+
+  @Test
+  void customContent() throws Exception {
+    CustomContent content =
+        ImmutableCustomContent.builder()
+            .id("id")
+            .customType("customType")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .build();
+    assertThat(deserialize(serialize(content), Content.class)).isEqualTo(content);
+  }
+
+  @Test
+  void unknownEventSerialization() throws Exception {
+    CustomEvent event =
+        ImmutableCustomEvent.builder()
+            .id(UUID.fromString("7385d1e6-3deb-440b-9008-a383e2de6e6c"))
+            .customType("weird")
+            .repositoryId("repo1")
+            .createdAt(Instant.parse("2023-04-25T13:02:05Z"))
+            .sentBy("Nessie")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .putProperty("boolean", true)
+            .build();
+    assertThat(serialize(event))
+        .isEqualTo(
+            "{"
+                + "\"id\":\"7385d1e6-3deb-440b-9008-a383e2de6e6c\","
+                + "\"type\":\"weird\","
+                + "\"repositoryId\":\"repo1\","
+                + "\"createdAt\":\"2023-04-25T13:02:05Z\","
+                + "\"sentBy\":\"Nessie\","
+                + "\"string\":\"foo\","
+                + "\"number\":123,"
+                + "\"boolean\":true"
+                + "}");
+  }
+
+  @Test
+  void unknownEventDeserialization() throws Exception {
+    CustomEvent event =
+        ImmutableCustomEvent.builder()
+            .id(UUID.fromString("7385d1e6-3deb-440b-9008-a383e2de6e6c"))
+            .customType("weird")
+            .repositoryId("repo1")
+            .createdAt(Instant.parse("2023-04-25T13:02:05Z"))
+            .sentBy("Nessie")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .putProperty("boolean", true)
+            .build();
+    assertThat(
+            deserialize(
+                "{"
+                    + "\"id\":\"7385d1e6-3deb-440b-9008-a383e2de6e6c\","
+                    + "\"type\":\"weird\","
+                    + "\"repositoryId\":\"repo1\","
+                    + "\"createdAt\":\"2023-04-25T13:02:05Z\","
+                    + "\"sentBy\":\"Nessie\","
+                    + "\"string\":\"foo\","
+                    + "\"number\":123,"
+                    + "\"boolean\":true"
+                    + "}",
+                Event.class))
+        .isEqualTo(event);
+  }
+
+  @Test
+  void unknownContentSerialization() throws Exception {
+    CustomContent content =
+        ImmutableCustomContent.builder()
+            .id("id")
+            .customType("weird")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .putProperty("boolean", true)
```

Comment:
Could you add a test case for a custom type with some data in `Content.getProperties()`?